src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2862,18 +2861,13 @@ ImageIcon getImage(URL base) {
             lengthMapping.put("cm", 28.3464f);
             lengthMapping.put("pc", 12f);
             lengthMapping.put("in", 72f);
-            int res = 72;
-            try {
-                res = Toolkit.getDefaultToolkit().getScreenResolution();
-            } catch (HeadlessException e) {
-            }
-            // mapping according to the CSS2 spec
-            w3cLengthMapping.put("pt", res / 72f);
-            w3cLengthMapping.put("px", 1f);
-            w3cLengthMapping.put("mm", res / 25.4f);
-            w3cLengthMapping.put("cm", res / 2.54f);
-            w3cLengthMapping.put("pc", res / 6f);
-            w3cLengthMapping.put("in", (float) res);
+            // mapping according to the CSS2.2 spec
+            w3cLengthMapping.put("pt", 1.3f); //1/72 of 1in
+            w3cLengthMapping.put("px", 1f); //1/96 of 1in
+            w3cLengthMapping.put("mm", 3.77952f); //1/10 of 1cm
+            w3cLengthMapping.put("cm", 37.7952f); //96px/2.54
+            w3cLengthMapping.put("pc", 16f); //1/6 of 1in
+            w3cLengthMapping.put("in", 96f); //96px
```

Comment:
> As for px, CSS defines, “1px is equal to 0.75pt”, then 1pt is 4/3px — this is where 1.3f comes from (=96/72).

This makes sense.

What makes me wonder is why the length mappings for `px` and `pt` without `W3C_LENGTH_UNITS` are exactly the opposite. In particular, it means that with W3C, `1pt > 1px`, while without W3C `1px > 1pt`.
That seems odd.

On the other hand, I don't know what the goal of those length mappings is, and they are not the sope of this PR.

Reply:
> 
> 
> > As for px, CSS defines, “1px is equal to 0.75pt”, then 1pt is 4/3px — this is where 1.3f comes from (=96/72).
> 
> This makes sense.
> 
> What makes me wonder is why the length mappings for `px` and `pt` without `W3C_LENGTH_UNITS` are exactly the opposite. In particular, it means that with W3C, `1pt > 1px`, while without W3C `1px > 1pt`.
> That seems odd.
> 
> On the other hand, I don't know what the goal of those length mappings is, and they are not the sope of this PR.

It's for historical reasons, I believe. JDK considers 1pt = 1px, as if the default screen resolution was 72dpi rather than 96dpi.